prov/verbs/src/verbs_checks.h

```diff
@@ -0,0 +1,346 @@
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _FI_VERBS_CHECKS_H
+#define _FI_VERBS_CHECKS_H
+#if HAVE_CONFIG_H
+#  include <config.h>
+#endif                          /* HAVE_CONFIG_H */
+
+#include <asm/types.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <poll.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <infiniband/ib.h>
+#include <infiniband/verbs.h>
+#include <rdma/rdma_cma.h>
+
+#include <rdma/fabric.h>
+#include <rdma/fi_cm.h>
+#include <rdma/fi_domain.h>
+#include <rdma/fi_prov.h>
+#include <rdma/fi_endpoint.h>
+#include <rdma/fi_rma.h>
+#include <rdma/fi_errno.h>
+
+#include <fi.h>
+#include <fi_enosys.h>
+
+static inline int fi_ibv_check_fabric_attr(const struct fi_fabric_attr *attr,
+                                           const struct fi_info *info)
+{
+    if (attr->name && strcmp(attr->name, info->fabric_attr->name)) {
+        FI_INFO(&fi_ibv_prov, FI_LOG_CORE, "Unknown fabric name\n");
+        return -FI_ENODATA;
+    }
+
+    if (attr->prov_version > info->fabric_attr->prov_version) {
+        FI_INFO(&fi_ibv_prov, FI_LOG_CORE, "Unsupported provider version\n");
+        return -FI_ENODATA;
+    }
+
+    return 0;
+}
+
+static inline int fi_ibv_check_domain_attr(const struct fi_domain_attr *attr,
+                                           const struct fi_info *info)
+{
+    if (attr->name && strcmp(attr->name, info->domain_attr->name)) {
+        FI_INFO(&fi_ibv_prov, FI_LOG_CORE, "Unknown domain name\n");
+        return -FI_ENODATA;
+    }
+
+    switch (attr->threading) {
+    case FI_THREAD_UNSPEC:
+    case FI_THREAD_ENDPOINT:
+        break;
+    case FI_THREAD_SAFE:
+    case FI_THREAD_FID:
+    case FI_THREAD_DOMAIN:
```

Comment:
Please make FI_THREAD_DOMAIN supported as well. Even if it isn't implemented we should pass the check in order to make MPICH/CH3/netmod/OFI work